### PR TITLE
Update to Unity 2019.4.24f1 and associated package updates

### PR DESCRIPTION
**Pull request by andybak**

### CODE REVIEW

**Comment by mikeage** [comment]
Thanks. That textmeshpro update is pretty important; because we don't have the UI option to "Import TMP Essential Resources", we have to do it manually in the CI file.

**Comment by mikeage** [comment]
This compilation might take a bit of time, since it'll need a new cache. Don't wait around for the binaries :-)

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -171,7 +171,7 @@ jobs:
           # version 2.0.1; must be updated if the version changes
```
Comment (mikeage) [inline .github/workflows/build.yml:171]: My mistake, can you please update this comment to say 2.1.4?

**Review by mikeage — CHANGES_REQUESTED** [review]

**Comment by mikeage** [comment]
Q2 seems happy!

File: .github/workflows/build.yml (hunk 1 above)
Comment (andybak) [reply .github/workflows/build.yml]: Done

**Review by mikeage — APPROVED** [review]

**Pull request merged**